src/platform/lemaker/bananapro.c

```diff
@@ -0,0 +1,116 @@
+/*
+	Copyright (c) 2016 CurlyMo <curlymoo1@gmail.com>
+
+  This Source Code Form is subject to the terms of the Mozilla Public
+  License, v. 2.0. If a copy of the MPL was not distributed with this
+  file, You can obtain one at http://mozilla.org/MPL/2.0/.
+*/
+
+#include <sys/mman.h>
+#include <unistd.h>
+#include <stdio.h>
+#include <fcntl.h>
+#include <errno.h>
+#include <string.h>
+#include <stdlib.h>
+#include <sys/ioctl.h>
+#include <signal.h>
+
+#include "../../soc/soc.h"
+#include "../../wiringx.h"
+#include "../platform.h"
+#include "bananapro.h"
+
+struct platform_t *bananapro = NULL;
+
+/* Banana Pro CON 6
+ *  only IO-0 ... IO-8 are supported by this software
```

Comment:
To be honest: I'm not so familiar with that soc related internals. My intention was to get it running to be used for one or two GPIOs on BananaPro, since I planned to setup pilight on BananaPro. For that reason I would like to limit the effort for my investigations on the soc. My intention is to provide a basement which can be reused by the community and extended if required.

Reply:
Allwinners are the easiest to decode:
http://dl.linux-sunxi.org/A20/A20%20User%20Manual%202013-03-22.pdf

Page 240